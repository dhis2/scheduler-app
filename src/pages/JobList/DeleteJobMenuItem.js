import React, { useState } from 'react'
import { string } from 'prop-types'
import { MenuItem } from '@dhis2/ui-core'
import { DeleteJobModal } from '../../components/Modal'

const DeleteJobMenuItem = ({ id }) => {
    const [showModal, setShowModal] = useState(false)

    return (
        <React.Fragment>
            <MenuItem
                dense
                destructive
                onClick={() => {
                    setShowModal(true)
                }}
                label="Delete"
            />
            {showModal && (
                <DeleteJobModal id={id} hideModal={() => setShowModal(false)} />
            )}
        </React.Fragment>
    )
}

DeleteJobMenuItem.propTypes = {
    id: string.isRequired,
}

export default DeleteJobMenuItem
