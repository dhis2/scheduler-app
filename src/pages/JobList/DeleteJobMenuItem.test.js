import React from 'react'
import { shallow, mount } from 'enzyme'
import { DumbDeleteJobMenuItem } from './DeleteJobMenuItem'

describe('<DeleteJobMenuItem>', () => {
    it('renders correctly', () => {
        const props = {
            id: 'id',
            showModal: () => {},
        }
        const wrapper = shallow(<DumbDeleteJobMenuItem {...props} />)

        expect(wrapper).toMatchSnapshot()
    })

    it('calls showModal when MenuItem is clicked', () => {
        const props = {
            id: 'id',
            showModal: jest.fn(),
        }
        const wrapper = mount(<DumbDeleteJobMenuItem {...props} />)

        wrapper.find('a').simulate('click')

        expect(props.showModal).toHaveBeenCalled()
    })
})
