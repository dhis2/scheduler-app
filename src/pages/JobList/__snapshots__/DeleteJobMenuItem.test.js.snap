// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteJobMenuItem> renders correctly 1`] = `
<Fragment>
  <MenuItem
    dataTest="dhis2-uicore-menuitem"
    dense={true}
    destructive={true}
    label="Delete"
    onClick={[Function]}
  />
</Fragment>
`;
